rockcraft/models/project.py

```diff
@@ -49,59 +53,17 @@
     _RunUser = Literal[tuple(SUPPORTED_GLOBAL_USERNAMES)] | None
 
 
-class Platform(craft_application.models.Platform):
-    """Rockcraft project platform definition."""
-
-    build_on: pydantic.conlist(str, unique_items=True, min_items=1) | None  # type: ignore[valid-type]
-    build_for: (  # type: ignore[valid-type]
-        pydantic.conlist(str, unique_items=True, min_items=1) | None  # type: ignore[reportInvalidTypeForm]
-    )
-
-    @pydantic.validator("build_for", pre=True)
-    @classmethod
-    def _vectorise_build_for(cls, val: str | list[str]) -> list[str]:
-        """Vectorise target architecture if needed."""
-        if isinstance(val, str):
-            val = [val]
-        return val
-
-    @pydantic.root_validator(skip_on_failure=True)
-    @classmethod
-    def _validate_platform_set(cls, values: Mapping[str, Any]) -> Mapping[str, Any]:
-        """Validate the build_on build_for combination."""
-        build_for: list[str] = values["build_for"] if values.get("build_for") else []
-        build_on: list[str] = values["build_on"] if values.get("build_on") else []
-
-        # We can only build for 1 arch at the moment
-        if len(build_for) > 1:
-            raise ValueError(
-                str(
-                    f"Trying to build a rock for {build_for} "
-                    "but multiple target architectures are not "
-                    "currently supported. Please specify only 1 value."
-                )
-            )
-
-        # If build_for is provided, then build_on must also be
-        if not build_on and build_for:
-            raise ValueError("'build-for' expects 'build-on' to also be provided.")
-
-        return values
-
-
-NAME_REGEX = r"^([a-z](?:-?[a-z0-9]){2,})$"
-"""
-The regex for valid names for rocks. It matches the accepted values for pebble
-layer files:
+PROJECT_NAME_REGEX = r"^([a-z](?:-?[a-z0-9]){2,})$"
```

Comment:
Since `max_length` is 40 below, could we do
```suggestion
PROJECT_NAME_REGEX = r"^([a-z](?:-?[a-z0-9]){2,39})$"
```

Reply:
actually I think I copied the `max_length` from craft-application - rocks don't have this restriction afaik. I'll remove it